client/init.js

```diff
@@ -87,17 +88,45 @@ addEventListener('load', function() {
 
 	function startWatchingPosition() {
 		navigator.geolocation.watchPosition(function(position) {
-			try {
-				comms.send({
-					event: 'friend-updated',
-					position: position
-				})
-			} catch (e) {
-				console.error(e)
-			}
+			sendMessage({
+				event: 'friend-updated',
+				position: position
+			})
 		})
 	}
 
+	function setupHiddenEvent(){
+		var visibility = new Visibility()
+		function handleVisibilityChange(){
+			var hiding = document[visibility.hidden]
+			if (hiding){
+				timeoutId = window.setTimeout(function(){
+					sendMessage({
+						event: 'friend-updated',
```

Comment:
So my thinking was to have two separate events. One is when they go hidden, notify users they're hidden. After the timeout is over itll broadcast that the users location is stale. I can figure out how to get it into the appropriate display